src/main/java/mods/railcraft/common/carts/EntityTunnelBore.java

```diff
@@ -454,9 +456,10 @@ public void onUpdate() {
 
                     if (boreLayer) {
                         boolean bored = boreLayer(targetPos, dir);
-                        if (bored)
-                            setDelay(LAYER_DELAY);
-                        else {
+                        if (bored) {
+                            int efficiency = EnchantmentHelper.getEnchantmentLevel(Enchantments.EFFICIENCY, getStackInSlot(0));
+                            setDelay(LAYER_DELAY / (int)(efficiency * .2 + 1));
```

Comment:
Does this still give enough time for gravity affected blocks like sand to fall before moving? That's the entire reason this delay even exists.

It would probably make more sense to move this to getLayerHardness() where the RailcraftConfig.boreMiningSpeedMultiplier() is applied.